log.go

```diff
@@ -4,207 +4,102 @@ import (
-}
-
-// Logger is a unified interface for all loggers.
+// Logger is an interface that should be implemented by all loggers wishing to participate
+// in the logger chain initialized by this package.
 type Logger interface {
-	Infof(format string, args ...interface{})
-	Warningf(format string, args ...interface{})
-	Errorf(format string, args ...interface{})
-	Fatalf(format string, args ...interface{})
-
+	Infof(string, ...interface{})
+	Warningf(string, ...interface{})
+	Errorf(string, ...interface{})
+
+	// Writer returns a logger's underlying io.Writer used to write log messages to.
+	//
+	// It may be, for example, the standard output for a console logger or a socket
+	// connection for a UDP logger.
 	Writer(Severity) io.Writer
+
+	// FormatMessage constructs and returns a final message that will go a logger's
+	// output channel.
+	FormatMessage(Severity, *CallerInfo, string, ...interface{}) string
+
+	String() string
 }
 
-// Logging configuration to be passed to all loggers during initialization.
-type LogConfig struct {
+// Config represents a configuration of an individual logger.
+type Config struct {
+	// Name is a logger's identificator used to instantiate a proper logger type
+	// from a config.
 	Name string
-}
 
-func (c LogConfig) String() string {
-	return fmt.Sprintf("LogConfig(Name=%v)", c.Name)
+	// Severity indicates the minimum severity a logger will be logging messages at.
+	Severity string
 }
 
-// SetSeverity sets current logging severity. Acceptable values are SeverityInfo, SeverityWarn, SeverityError, SeverityFatal
-func SetSeverity(s Severity) {
-	currentSeverity.Set(s)
+func (c Config) String() string {
+	return fmt.Sprintf("Config(Name=%v, Severity=%v)", c.Name, c.Severity)
 }
 
-// GetSeverity returns currently set severity.
-func GetSeverity() Severity {
-	return currentSeverity
+// Init initializes the logging package with the provided loggers.
+func Init(l ...Logger) {
+	for _, logger := range l {
+		loggers = append(loggers, logger)
+	}
 }
 
-// Logging initialization, must be called at the beginning of your cool app.
-func Init(logConfigs []*LogConfig) error {
-	loggers := []Logger{}
-	for _, config := range logConfigs {
+// InitWithConfig instantiates loggers based on the provided configs and initializes
+// the package with them.
+func InitWithConfig(configs ...Config) error {
+	for _, config := range configs {
 		l, err := NewLogger(config)
 		if err != nil {
 			return err
 		}
 		loggers = append(loggers, l)
 	}
-	logger.setLoggers(loggers...)
 	return nil
 }
 
-// Make a proper logger from a given configuration.
-func NewLogger(config *LogConfig) (Logger, error) {
+// NewLogger makes a proper logger from the given configuration.
+func NewLogger(config Config) (Logger, error) {
 	switch config.Name {
-	case "console":
+	case ConsoleLoggerName:
 		return NewConsoleLogger(config)
-	case "syslog":
+	case SysLoggerName:
 		return NewSysLogger(config)
+	case UDPLoggerName:
```

Comment:
I would also define all three in one const block in this file, to generate better godocs:

```go
// Supported log types.
const (
    Console = "console"
    Syslog  = "syslog"
    UDPLog  = "udplog"
)
```